### PR TITLE
Corrected latext for unevaluated Mul

### LINKED ISSUE #21094: Incorrect latex with unevaluated Mul
`latex(Mul(6, S.Half, evaluate=False))` gives `6 1/2`, so there is no `numbersep`.

### PR DESCRIPTION
**Pull request by iam-abbas**
#### References to other Issues or PRs

Fixes #21094

#### Brief description of what is fixed or changed
Fixed incorrect latex output for unevaluated `Mul` while using fractions.  

Before
```python
>>> latex(Mul(1, S.Half, evaluate=False))
'1  \\frac{1}{2}'
```
Now
```python
>>> latex(Mul(1, S.Half, evaluate=False))
'1 \\cdot  \\frac{1}{2}'
```
Also updated the test cases for the same bug. 

#### Release Notes

- printing
   - Fixed issue with printing latex of unevaluated `Mul` with fractions

### CODE REVIEW

File: sympy/printing/latex.py
Comment (namannimmo10) [inline sympy/printing/latex.py:532]: Thanks for the fix! Can you add more tests that hit this line?

Comment (iam-abbas) [reply sympy/printing/latex.py:532]: I have added extra tests covering negative expressions and exponents

**Review by namannimmo10 — APPROVED** [review]
LGTM

**Comment by iam-abbas** [comment]
Is there anything else that needs to be done?

**Comment by namannimmo10** [comment]
I believe this can go in now! Let's wait for someone else's review. @sylee957

File: sympy/printing/latex.py
Comment (sylee957) [reply sympy/printing/latex.py:532]: Why does this had to use `str(term)` than `term_tex`?

Comment (iam-abbas) [reply sympy/printing/latex.py:532]: term_tex for fractions has `\frac{}{}` and the regex code is running a `match()` for strings that start with `{[0-9` so the `\frac` never gets picked up because `match()` only matches from the beginning and you can't come up with a regex (very complicated) for `\frac` because of the backslash, hence change the input itself to term rather than latex turn because in the end the regex is just checking for number

File: sympy/printing/latex.py
Comment (oscarbenjamin) [inline sympy/printing/latex.py:118]: If the regex is just `[0-9]` then isn't that equivalent to `s.isascii() and s.isdecimal()`?

It would seem simpler to use that than a regex.

Comment (iam-abbas) [reply sympy/printing/latex.py:118]: `isascii` returns true for all ascii characters and `isdecimal` returns true for unicode

Comment (oscarbenjamin) [reply sympy/printing/latex.py:118]: Right, so:
```
if s.isascii() and s.isdecimal():
```

Comment (iam-abbas) [reply sympy/printing/latex.py:118]: Sorry didn't interpret that correctly. Yes, you're right. It will work

File: sympy/printing/latex.py
Comment (sylee957) [reply sympy/printing/latex.py:532]: I still get the impression that this is an workaround and it can't handle the second case

```
>>> latex(Mul(2, Mul(Rational(3, 4), evaluate=False), 5, evaluate=False))
'2 \\cdot \\frac{3}{4} \\cdot 5'
>>> latex(Mul(2, Mul(Rational(3, 4), 5, evaluate=False), evaluate=False))
'2 \\frac{3}{4} \\cdot 5'
```

File: sympy/printing/latex.py
Comment (sylee957) [reply sympy/printing/latex.py:118]: Why should we worry about unicode?

File: sympy/printing/latex.py
Comment (iam-abbas) [reply sympy/printing/latex.py:532]: What should be the expected output for the second case?

File: sympy/printing/latex.py
Comment (oscarbenjamin) [reply sympy/printing/latex.py:118]: I don't know. I was just suggesting equivalent code.

File: sympy/printing/latex.py
Comment (sylee957) [reply sympy/printing/latex.py:532]: My opinion is that the second should have the parenthesis between the second and the third argument
and also some consistent way about applying `cdot`
I have no clear idea about how 'between number' was defined in the past and how consistent the rule was, but I think that it's time to clarify them before jumping on a implementation that may complicate the printing code.

**Comment by oscargus** [comment]
Closing and reopening to trigger the tests. I think this is a step in the right direction.

**Comment by oscargus** [comment]
Although this is probably not the final solution to the issue of printing unevaluated `Mul`, it appears to be a step in the right direction. I will merge in a while if I do not here any objections. (As there is quite some time before this will get into a release, there is also time to find a better solution if something shows up.)

**Comment by iam-abbas** [comment]
Is there anything else that can be added a quick change?

**Comment by oscargus** [comment]
I think it is fine as is. Probably one should really start from the beginning when it comes to unevaluated multiplications. Right now there is some quite well thought through code how to print it if it is evaluated and then it appears that one tries to revert that later. A better approach may be to have two completely separate paths for evaluated and unevaluated multiplications, but I guess that it is not straightforward to know if an expression is evaluated or not.

Thanks for the contribution!

**Pull request merged**
